### ISSUE TITLE
Template for GeoAxis 

### ISSUE THREAD

**Issue by SimonDanisch** [MakieOrg/GeoMakie.jl#92]
Until https://github.com/JuliaPlots/Makie.jl/pull/1355 is merged, we can use something like this for a GeoAxis:

```julia
using GLMakie, GeoMakie
using GeoMakie.GeoInterface
using GeoMakie.GeoJSON
using Proj4

struct GeoAxis
    axis::LScene
    kw
end

function GeoAxis(args...; source = "+proj=longlat +datum=WGS84", dest = "+proj=natearth2", kw...)
    trans = Proj4.Transformation(source, dest, always_xy=true)
    ptrans = Makie.PointTrans{2}(trans)
    ax = LScene(args...; scenekw=(show_axis=false, camera=cam2d!))
    # TODO, pass this to scenekw?!
    ax.scene.transformation.transform_func[] = ptrans
    # here you can draw in any axis grids etc
    return GeoAxis(ax, kw)
end

function bb_transformed(plot::AbstractPlot)
    # Relies on Makie#master 
    # There's definitely also a way to make thise work on tagged Makie...
    bb_ref = Base.RefValue(Rect3f())
    points = Makie.point_iterator(plot)
    t = Makie.transformation(plot)
    model = Makie.model_transform(t)
    trans_func = t.transform_func[]
    Makie.foreach_transformed(points, model, trans_func) do point
        Makie.update_boundingbox!(bb_ref, point)
    end
    return bb_ref[]
end


function Makie.plot!(
        ta::GeoAxis, P::Makie.PlotFunc,
        attributes::Makie.Attributes, args...)
    # Do PlotFunc (Heatmap, Surface, etc) specific overloads here
    p = Makie.plot!(ta.axis.scene, P, attributes, args...)
    # Do custom, GeoAxis specific things like updating the transformed limits
    update_cam!(ta.axis.scene, bb_transformed(p))
    return p
end

# just part of overloading the pipeline... The above is the more generic one!
function Makie.plot!(P::Makie.PlotFunc, ax::GeoAxis, args...; kw_attributes...)
    Makie.plot!(ax, P, Attributes(kw_attributes), args...)
end

fig = Figure()
ax = GeoAxis(fig[1, 1])

lats = -90:10.0:90
lons = -180:10.0:180
lons = collect(lons)
lons[end] = prevfloat(lons[end])  # avoid PROJ wrapping 180 to -180
field = [exp(cosd(l)) + 3(y/90) for l in lons, y in lats]

p = wireframe!(ax, lons, lats, field, color=(:gray, 0.2), transparency=true)
fig
```

**Comment by asinghvi17** [comment]
Is there a way to make `GeoAxis` inherit `Axis`'s attributes?  That would make things a lot easier and more observable-friendly, too.  Right now, for example, if you want to change the transformation post-axis creation, it's a pain.

**Comment by asinghvi17** [comment]
I think the basic structure of this is implemented in #207 and before.
